src/physics/physicsman.cpp

```diff
@@ -83,4 +88,58 @@ void PhysicsManager::remove(btActionInterface *actionInterface) {
 	_world->removeAction(actionInterface);
 }
 
+const CastResult PhysicsManager::raycastStatic(glm::vec3& from, glm::vec3& to) {
```

Comment:
Hmmm... While I can indeed do that, that seems to go against the reason that `const` was put there in the first place - to signal that CastResult is not supposed to be changed. As far as I understand, making the method `const` will not make return type const in this case, because we create it inside the function body 
```c++
// src/physics/physicsman.cpp:102
return CastResult{
	true,
	rayHitPoint,
	rayHitPoint
};
// src/physics/physicsman.cpp:109
return CastResult{
	false,
	glm::zero<glm::vec3>(),
	glm::zero<glm::vec3>()
};
``` 
and not return some field of the `PhysicsManager` object (which *would* carry the const modifier of the parent object). However, it is a good idea to mark casting methods `const` as well since they do not change `PhysicsManager`.

Reply:
Yeah, but you return the CastResult by value, which means the const is practically meaningless, since it can be copied. What you did already is enough to ensure it stays immutable.